### PR TITLE
Revised deployment topic to include Visual Studio

### PR DESCRIPTION
**Pull request by rpetrusha**
# Revised deployment topic to include Visual Studio

## Summary

This addresses a customer complaint that it did not appear to be possible to deploy a self-contained app outside of the command line.

## Suggested Reviewers

@blackdwarf  @mairaw

### CODE REVIEW

File: docs/core/deploying/index.md [hunk 1]
```diff
@@ -39,86 +39,83 @@ There are also a few disadvantages:
...
+
+Deploying a framework-dependent deployment with no third-party dependencies simply involves building, testing, and publishing the app. A simple example written in C# illustrates the process. The example explains how to create a framework-dependent deployment by using the [dotnet utility](../tools/dotnet.md) from the command line, as well as by using Visual Studio 2017. If you are working from the command line, you can use a program editor of your choice. If your program editor is Visual Studio Code, you can open a command console inside your Visual Studio Code environment by selecting **View**, **Integrated Terminal**.
+
+1. Create the project.
+
+   **From the command line:** Create a directory for your project, and from the command line, type [dotnet new console](../tools/dotnet-new.md) to create a new C# console project.
+
+   **In Visual Studio 2017:** Select **File**, **New**, **Project**. In the **Add New Project** dialog, select **.NET Core** in the **Installed** project types pane, and select the **Console App (.NET Core)** template in the center pane. Enter a project name, such as `FDD`, in the **Name** text box, and select the **OK** button.
```
Comment (mairaw) [inline docs/core/deploying/index.md]: MSFT style guide: Use > to separate sequential steps

File: docs/core/deploying/index.md [hunk 2]
```diff
@@ -39,86 +39,83 @@ There are also a few disadvantages:
...
+
+1. Open the `Program.cs` file in your editor, and replace the auto-generated code with the following code. It prompts the user to enter text, and then displays the individual words entered by the user. It uses the regular expression `\w+` to separate the words in the input text.
```
Comment (mairaw) [inline docs/core/deploying/index.md]: style file names as italic

**Comment by mairaw** [comment]
I've been thinking about this topic and wondering whether we should separate the theory from the practice. And then we could have a topic that is totally done through VS vs. one that is all CLI. Thoughts?

File: docs/core/deploying/index.md (hunk 1 above)
Comment (rpetrusha) [inline docs/core/deploying/index.md]: Fixed.

File: docs/core/deploying/index.md (hunk 2 above)
Comment (rpetrusha) [inline docs/core/deploying/index.md]: Fixed.

**Comment by rpetrusha** [comment]
I can see the advantages of each approach, though my own preference here is for a single, longer topic, as long as the steps for CLI/VS are clearly differentiated, and I think that they are here -- or at least, I've tried to make them clearly distinct. At the same time, I have to admit that I tend to have a bias toward longer topics. @blackdwarf, do you have a sense of whether this should remain one single topic, or be broken up into one conceptual overview, one CLI how-to topic, and one VS topic?

**Comment by rpetrusha** [comment]
@mairaw In thinking more about it, although I'd prefer not to essentially duplicate content in separate Visual Studio and CLI topics, I agree that the best thing to do here would be to separate the two implementation topics from the conceptual topic. I'll do that, and in the meantime, I'll add the WIP label.

**Comment by rpetrusha** [comment]
@mairaw @blackdwarf I've separated the conceptual portion of the topic from the CLI and Visual Studio examples, which I've put in separate topics.

**Comment by GuardRex** [comment]
@rpetrusha Is it ok if I review this and we make a move to get it in there. I :heart: SCD ... It is my favorite deployment type.

**Comment by rpetrusha** [comment]
@GuardRex, yes, by all means!

**Comment by GuardRex** [comment]
@rpetrusha I noticed that you moved content from the initial doc into the two examples docs unchanged/uncorrected. I see you were just getting the content in the right spot on this pass. Should I skip the nits for now? ... someone can make a grammar/style pass later.

**Comment by rpetrusha** [comment]
@GuardRex, no, review it thoroughly. My intent was to do more than get the content in the right places, it was to get the content correct.

File: docs/core/deploying/index.md [hunk 3]
```diff
@@ -13,17 +13,17 @@ ms.assetid: da7a31a0-8072-4f23-82aa-8a19184cb701
...
 
 - Framework-dependent deployment. As the name implies, framework-dependent deployment (FDD) relies on a shared system-wide version of .NET Core to be present on the target system. Because .NET Core is already present, your app is also portable between installations of .NET Core. Your app contains only its own code and any third-party dependencies that are outside of the .NET Core libraries. FDDs contain .dll files that can be launched by using the [dotnet utility](../tools/dotnet.md) from the command line. For example, `dotnet app.dll` runs an application named `app`.
 
 - Self-contained deployment. Unlike FDD, a self-contained deployment (SCD) does not rely on any shared components to be present on the target system. All components, including both .NET Core libraries and the .NET Core runtime, are included with the application and are isolated from other .NET Core applications. SCDs include an executable (such as `app.exe` on Windows platforms for an application named `app`), which is  a renamed version of the platform-specific .NET Core host, and a .dll file (such as `app.dll`), which is the actual application.
```
Comment (GuardRex) [inline docs/core/deploying/index.md]: * does not :point_right: doesn't
* "rely on any shared components to be present on the target system" :point_right: "rely on the presence of shared components on the target system"
* "both .NET Core libraries" :point_right: "both the .NET Core libraries"
* .dll :point_right: italics?
* \`app.dll\` :point_right: italics

File: docs/core/deploying/index.md [hunk 4]
```diff
@@ -13,17 +13,17 @@ ms.assetid: da7a31a0-8072-4f23-82aa-8a19184cb701
...
 
-## Framework-dependent deployments (FDD) ##
+## Framework-dependent deployments (FDD)
 
 For an FDD, you deploy only your app and any third-party dependencies. You do not have to deploy .NET Core, since your app will use the version of .NET Core that's present on the target system. This is the default deployment model for .NET Core apps.
```
Comment (GuardRex) [inline docs/core/deploying/index.md]: "do not" :point_right: "don't"

File: docs/core/deploying/index.md [hunk 5]
```diff
@@ -13,17 +13,17 @@ ms.assetid: da7a31a0-8072-4f23-82aa-8a19184cb701
...
 
-### Why create a framework-dependent deployment? ###
+### Why create a framework-dependent deployment?
 
 Deploying an FDD has a number of advantages:
```
Comment (GuardRex) [inline docs/core/deploying/index.md]: In the list below ...

First bullet:
* "do not" :point_right: "don't"
* "You do not have to define the target operating systems that your .NET Core app will run on in advance." :point_right: "You don't need to define in advance the target operating systems where your .NET Core app will run."
* "PE file format" isn't defined (it's linked, but not defined)
* The 2nd sentence somewhat restates the info in the first sentence

Second bullet:
* "You only have to deploy" :point_right: "You only deploy"

File: docs/core/deploying/index.md [hunk 6]
```diff
@@ -39,86 +39,11 @@ There are also a few disadvantages:
 
 - It is possible for the .NET Core runtime and libraries to change without your knowledge in future releases. In rare cases, this may change the behavior of your app.
```
Comment (GuardRex) [inline docs/core/deploying/index.md]: * "It is" :point_right: "It's"
* Vaguely feels like an example is needed to flesh out the last remark "in rare cases ..."

File: docs/core/deploying/index.md [hunk 7]
```diff
@@ -39,86 +39,11 @@ There are also a few disadvantages:
...
 
-### Deploying a framework-dependent deployment ###
+## Self-contained deployments (SCD)
 
-Deploying a framework-dependent deployment with no third-party dependencies simply involves building, testing, and publishing the app. A simple example written in C# illustrates the process. The example uses the [dotnet utility](../tools/dotnet.md) from the command line; however, you can also use a development environment, such as Visual Studio or Visual Studio Code, to compile, test, and publish the example.
+For a self-contained deployment, you deploy not only your app and any third-party dependencies, but the version of .NET Core that you build your app with. Creating an SCD does not, however, include the [native dependencies of .NET Core](https://github.com/dotnet/core/blob/master/Documentation/prereqs.md) itself on various platforms (for example, OpenSSL on macOS), so these need to be installed before running the application.
```
Comment (GuardRex) [inline docs/core/deploying/index.md]: > dependencies, but the version

remove comma

... or another way to state this that kills the sentence-ending preposition is ...

> For a self-contained deployment, you deploy your app and any included third-party dependencies along with the version of .NET Core used to build the app.

Next sentence:
* In that position, "however" seems *more formal* than just starting with the word.
* "does not" :point_right: "doesn't"
* I would drop the word "itself"
* "need to be installed" :point_right: "must be present" (the dev doesn't need to do it if they're already there)
* application :point_right: app

> However, creating an SCD doesn't include ...

File: docs/core/deploying/index.md [hunk 8]
```diff
@@ -134,177 +59,19 @@ It also has a number of disadvantages:
...
-    <TargetFramework>netstandard1.6</TargetFramework>
-    <RuntimeIdentifiers>win10-x64;osx.10.11-x64</RuntimeIdentifiers>
-  </PropertyGroup>
-  <ItemGroup>
-    <PackageReference Include="Microsoft.NETCore.Runtime.CoreCLR" Version="1.0.2" />
-    <PackageReference Include="Microsoft.NETCore.DotNetHostPolicy" Version="1.0.1" />
-  </ItemGroup>
-</Project>
-```
+# See also
```
Comment (GuardRex) [inline docs/core/deploying/index.md]: Consistency: A blank line after the section heading

File: samples/snippets/core/deploying/deployment-example.cs [hunk 9]
```diff
@@ -0,0 +1,36 @@
+using System;
```
Comment (GuardRex) [inline samples/snippets/core/deploying/deployment-example.cs]: As I stated up at the start, this isn't necessary to teach these concepts or show how the tooling works to create these deployment types. Therefore, I suggest removing this and just going with a good 'ole default "Hello World!" console app.

**Review by guardrex — CHANGES_REQUESTED** [review]
@rpetrusha There are a handful of purely writing style suggestions that will be easy for you to skip if you don't like them. The vast majority of these comments are nit-picky little grammar-ish thingies. There are a few gems in there. Although text is duplicated between two docs, I tried to catch every instance in both docs to make it easier for you to run down the suggestions. You don't need to see a suggestion in one and skip to the other ... my comment re-appears in the other at the right spot (I hope).

**Comment by GuardRex** [comment]
A passing note on web apps + 👣 small footprint 👣 SCD: The tracking issue is https://github.com/dotnet/cli/issues/5769. Nobody has asked beyond my tracking, so perhaps nothing needs to be noted at this time in the small footprint section. I'm going to check again this weekend and at least see if the `web.config` fix works for me, and I'll see if the other two issues are still problems.

**Comment by rpetrusha** [comment]
@GuardRex, I've finally revised this based on your comments. Thanks for the extremely through review!

File: docs/core/deploying/deploying-with-cli.md [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+
+   Create a directory for your project and make it your current directory.
+
+1. Create the project.
+
+   From the command line, type [dotnet new console](../tools/dotnet-new.md) to create a new C# console project in that directory.
+
+1. Add the application's source code.
+
+   Open the *Program.cs* file in your editor and replace the auto-generated code with the following code. It prompts the user to enter text, and then displays the individual words entered by the user. It uses the regular expression `\w+` to separate the words in the input text.
```
Comment (GuardRex) [inline docs/core/deploying/deploying-with-cli.md]: Remove comma after "text" (I suggest dropping "then" as well)

> It prompts the user to enter text and displays the individual words entered by the user.

File: docs/core/deploying/deploying-with-cli.md [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+
+   Create a directory for your project, and make it your current directory.
+
+1. Create the project.
+
+   From the command line, type [dotnet new console](../tools/dotnet-new.md) to create a new C# console project in that directory.
+
+1. Add the application's source code.
+
+   Open the *Program.cs* file in your editor and replace the auto-generated code with the following code. It prompts the user to enter text and displays the individual words entered by the user. It uses the regular expression `\w+` to separate the words in the input text.
```
Comment (GuardRex) [inline docs/core/deploying/deploying-with-cli.md]: Remove comma after "text" (I suggest dropping "then" as well)

> It prompts the user to enter text and displays the individual words entered by the user.

File: docs/core/deploying/deploying-with-vs.md [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+
+Deploying a self-contained deployment with no third-party dependencies involves creating the project, modifying the *csproj* file, building, testing, and publishing the app.  A simple example written in C# illustrates the process. 
+
+1. Create the project.
+
+   Select **File** > **New** > **Project**. In the **Add New Project** dialog, select **.NET Core** in the **Installed** project types pane, and select the **Console App (.NET Core)** template in the center pane. Enter a project name, such as `SCD`, in the **Name** text box, and select the **OK** button.
+
+1. Add the application's source code.
+ 
+   Open the `Program.cs` file in your editor, and replace the auto-generated code with the following code. It prompts the user to enter text, and then displays the individual words entered by the user. It uses the regular expression `\w+` to separate the words in the input text.
```
Comment (GuardRex) [inline docs/core/deploying/deploying-with-vs.md]: \*Program.cs\*
Remove comma after "text" (I'd suggest dropping "then" as well)

> It prompts the user to enter text and displays the individual words entered by the user.

File: docs/core/deploying/deploying-with-vs.md [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+
+## Framework-dependent deployment with third-party dependencies
+
+Deploying a framework-dependent deployment with one or more third-party dependencies requires that any dependencies be available to your project. The following additional steps are required before you can build your app:
+
+1. Use the **NuGet Package Manager** to add a reference to a NuGet package to your project; and if the package is not already available on your system, install it. To open the package manager, select **Tools** > **NuGet Package Manager** > **Manage NuGet Packages for Solution**.
+
+1. Confirm that Newtonsoft.Json is installed on your system and, if it is not, install it. The **Installed** tab lists NuGet packages installed on your system. If Newtonsoft.Json is not listed there, select the **Browse** tab and enter 'Newtonsoft.Json' in the search box. Select Newtonsoft.Json and, in the right pane, select your project before selecting **Install**.
+
+1. If Newtonsoft.Json is already installed on your system, add it to your project by selecting your project in the right pane of hte **Manage Packages for Solution** tab.
```
Comment (GuardRex) [inline docs/core/deploying/deploying-with-vs.md]: \`Newtonsoft.Json\`

File: docs/core/deploying/index.md [hunk 14]
```diff
@@ -1,124 +1,49 @@
...
 
-- Self-contained deployment. Unlike FDD, a self-contained deployment (SCD) does not rely on any shared components to be present on the target system. All components, including both .NET Core libraries and the .NET Core runtime, are included with the application and are isolated from other .NET Core applications. SCDs include an executable (such as `app.exe` on Windows platforms for an application named `app`), which is  a renamed version of the platform-specific .NET Core host, and a .dll file (such as `app.dll`), which is the actual application.
+- Self-contained deployment. Unlike FDD, a self-contained deployment (SCD) doesn't rely on the presence of shared components on the target system. All components, including both the .NET Core libraries and the .NET Core runtime, are included with the application and are isolated from other .NET Core applications. SCDs include an executable (such as `app.exe` on Windows platforms for an application named `app`), which is  a renamed version of the platform-specific .NET Core host, and a *.dll* file (such as *app.dll*), which is the actual application.
```
Comment (GuardRex) [inline docs/core/deploying/index.md]: \*app.exe\*

**Review by guardrex — APPROVED** [review]

**Comment by GuardRex** [comment]
@rpetrusha Looks great ... just a handful of minor touches and :shipit:

**Comment by rpetrusha** [comment]
Thanks, @GuardRex. I've incorporated all of your remaining changes; I'll merge when the build finishes.

**Pull request merged**